private/buf/buflsp/completion.go

```diff
@@ -208,19 +208,61 @@ func completionItemsForDef(ctx context.Context, file *file, declPath []ast.DeclA
 // completionItemsForImport returns completion items for import declarations.
 //
 // Suggest all importable files.
-func completionItemsForImport(ctx context.Context, file *file) []protocol.CompletionItem {
+func completionItemsForImport(ctx context.Context, file *file, declImport ast.DeclImport, position protocol.Position) []protocol.CompletionItem {
 	file.lsp.logger.DebugContext(ctx, "completion: import declaration", slog.Int("importable_count", len(file.importToFile)))
 
+	positionLocation := file.file.InverseLocation(int(position.Line)+1, int(position.Character)+1, positionalEncoding)
+	offset := positionLocation.Offset
+
+	importPathSpan := declImport.ImportPath().Span()
+	importPathText := importPathSpan.Text()
+	if importPathSpan.Start > offset || offset > importPathSpan.End {
```

Comment:
@stefanvanburen I think we can follow this pattern hopefully in other completions:
- Get the AST node for the current position
- Figure out the context of the position in the current expression
- Suggest completions that match the current expression